autonomy/analyse/abci/app_spec.py

```diff
@@ -338,6 +343,51 @@ def abci_to_dfa(cls, abci_app_cls: Any, label: str = "") -> "DFA":
         )
 
 
+event_pattern = re.compile(r"Event\.(\w+)", re.DOTALL)
+
+
+def _check_unreferenced_events(abci_app_cls: AbciApp) -> None:
+    """Checks for unreferenced events in the AbciApp.
+
+    Checks that events defined in the AbciApp transition function are referenced
+    in the source code of the corresponding rounds or their superclasses. Note that
+    the function simply checks references in the "raw" source code of the rounds and
+    their (non builtin) superclasses. Therefore, it does not do any kind of static
+    analysis on the source code, nor checks for actual reachability of a return
+    statement returning such events.
+
+    :param abci_app_cls: AbciApp to check unreferenced events.
+    :raises DFASpecificationError: If there are unfererenced events in the AbciApp.
+    """
+
+    error_strings = []
+    timeout_events = {k.name for k in abci_app_cls.event_to_timeout.keys()}
+
+    for round_cls, round_transitions in abci_app_cls.transition_function.items():
+        trf_events = {
+            str(e).rsplit(".", 1)[1] for e in round_transitions
+        } - timeout_events
```

Comment:
Why do we remove timeout events?

Reply:
I assume because it is trivially present in all cases?

Reply:
Because they are not returned explicitly in the source code of the rounds.